OpenRA.Mods.Cnc/FileFormats/WsaReader.cs

```diff
@@ -0,0 +1,135 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.IO;
+using OpenRA.Video;
+
+namespace OpenRA.Mods.Cnc.FileFormats
+{
+	public class WsaReader : IVideo
```

Comment:
I often find it very helpful to have a reference in the code to the source of knowledge, especially with obscure file formats like this.
Something like 
```suggestion
namespace OpenRA.Mods.Cnc.FileFormats
{
	// Implementation based on:
	// https://github.com/ultraq/redhorizon/blob/main/redhorizon-classic/source/nz/net/ultraq/redhorizon/classic/filetypes/wsa/WsaFile.groovy
	// https://moddingwiki.shikadi.net/wiki/Westwood_WSA_Format
	public class WsaReader : IVideo
```

Reply:
Those are not very citeable and I didn't base upon them.

Reply:
Linked it in the [wiki](https://github.com/OpenRA/OpenRA/wiki/Reverse-engineering/_compare/76536fed9dc79a71825ebae63d507d562cebb624...5c0c18d24d2d43940e11307066bfa33667b63cec) instead.